src/ToDo/Business/ITaskService.cs

```diff
@@ -9,4 +9,6 @@ public interface ITaskService
 	Task UpdateAsync(ToDoTask task, CancellationToken ct);
 
 	Task DeleteAsync(ToDoTask task, CancellationToken ct);
+
+	ValueTask<IImmutableList<ToDoTask>> GetAllAsync(string displayName = "", CancellationToken ct = default);
```

Comment:
What was the intent behind returning an iimutablelist? Could you just return either ToDoTask[] or IEnumerable<ToDoTask>?

Reply:
Yes, I'll change to return IEnumerable. the IInmutableList it's not neccessary.

Reply:
Leave this for the moment - I realised that the other methods on the services are also immutable lists